diff --git a/CMR2Decomp/Race.cpp b/CMR2Decomp/Race.cpp
index bcd6d94..e06ffcc 100644
--- a/CMR2Decomp/Race.cpp
+++ b/CMR2Decomp/Race.cpp
@@ -925,8 +925,8 @@ void Race_QueuePlayerCallout(unsigned int player, BYTE callId, BYTE prevCallId,
 
     for (i = 0; i < 5; i++) {
         if ((g_raceCallRecords[player * 5 + i].flags & 0x100) == 0) {
-            g_raceCallRecords[player * 5 + i].field_0x0 = id;
             g_raceCallRecords[player * 5 + i].flags = (g_raceCallRecords[player * 5 + i].flags & 0xfffffd32) | 0x132;
+            g_raceCallRecords[player * 5 + i].field_0x0 = id;
             if (prev != 0) {
                 g_raceCallRecords[player * 5 + i].field_0x4 = prev;
                 if (Race_DecodeCallSlotIcon(prev, &texture, &rect, &flag, colour) != 0)
