diff --git a/CMR2Decomp/FrontendDraw.cpp b/CMR2Decomp/FrontendDraw.cpp
index 513f4f2..f864123 100644
--- a/CMR2Decomp/FrontendDraw.cpp
+++ b/CMR2Decomp/FrontendDraw.cpp
@@ -375,8 +375,8 @@ void FrontendDraw_MenuList(Menu *pMenu, char *title, int y, int xOffset, int fir
 
     count = 0;
     hasTitle = 0;
-    dst.y = 0;
     dst.x = (int)(g_pGraphics->resX * 100) / 640;
+    dst.y = 0;
     dst.w = CFrontend::m_pAr640ATexture->width;
     dst.h = CFrontend::m_pAr640ATexture->height;
     for (i = 0; i < pMenu->itemCount; i++) {
